FordLabs/retroquest
# Team Name Recovery Page

## Overview
* This PR adds a team name recovery page that is linked to from the request password reset page.
* In this PR I also moved the endpoints that are sending emails into an EmailController as the team service was getting hard to follow which endpoints were responsible for teams and which were responsible for other things.

### Demo
<img width="562" alt="Screen Shot 2022-10-07 at 12 08 01 PM" src="https://user-images.githubusercontent.com/17144844/194598828-e6c44501-a3cd-47fc-a12f-1eee95a7c107.png">

<img width="570" alt="Screen Shot 2022-10-07 at 10 58 36 AM" src="https://user-images.githubusercontent.com/17144844/194598881-de7663da-8534-4fff-b359-12a47b696208.png">


<img width="503" alt="Screen Shot 2022-10-07 at 12 01 15 PM" src="https://user-images.githubusercontent.com/17144844/194598836-e2d5c856-7984-46bd-8e71-33be3cf7199a.png">

## Testing Instructions
1) Go to `http://localhost:3000/request-password-reset` and click the `I don't remember my team name` link.
2) Type in an incorrect email and submit. Ensure there is an error message saying that that email is not associated with any teams.
3) Type in an email address that is associated with at least one team. Submit form.
4) Confirm that there is a confirmation message saying to check your mail, what email address you just typed in, and what email address to expect the email to come from.
5) Check that an email has been sent (this part can only be tested in dev) and that it includes all team names that email is associated with.
